### PR TITLE
xds: Validate NPHDS updates before upserting

### PR DESCRIPTION
**Pull request by joestringer**
This should catch recent errors such as the one reported in #3825.

@rlenglet @jrajahalme  I played around a bit with this locally but couldn't reproduce #3825, so not sure what conditions are necessary for it. In the mean time we can certainly add some better validation / logging as per this commit.

### CODE REVIEW

**Comment by joestringer** [comment]
test-me-please

File: pkg/envoy/resources.go [hunk 1]
```diff
@@ -123,6 +132,10 @@ func (cache *NPHDSCache) handleIPDelete(npHost *envoyAPI.NetworkPolicyHosts, pee
 		} else {
 			npHost.HostAddresses = append(npHost.HostAddresses[0:targetIndex], npHost.HostAddresses[targetIndex+1:]...)
 		}
+		if err := npHost.Validate(); err != nil {
+			scopedLog.WithError(err).Warning("Could not validate NPHDS delete message")
```
Comment (rlenglet) [inline pkg/envoy/resources.go]: NPHDS delete message -> NPHDS resource after deleting host

File: pkg/envoy/resources.go [hunk 2]
```diff
@@ -85,6 +86,12 @@ func (cache *NPHDSCache) OnIPIdentityCacheChange(
 		}
 		npHost.HostAddresses = append(npHost.HostAddresses, ipIDPair.IP.String())
 		sort.Strings(npHost.HostAddresses)
+		if err := npHost.Validate(); err != nil {
+			scopedLog.WithError(err).WithFields(logrus.Fields{
+				logfields.XDSResource: npHost,
+			}).Warning("Could not validate NPHDS update message")
```
Comment (rlenglet) [inline pkg/envoy/resources.go]: NPHDS update message -> NPHDS resource after upserting host

**Review by rlenglet — APPROVED** [review]
LGTM! Nit re: the log messages.

File: pkg/envoy/resources.go (hunk 1 above)
Comment (joestringer) [reply pkg/envoy/resources.go]: NPHDS implies host at the moment, and I'll need to update the "host" part almost immediately with my new CIDR PR. I can 's/message/resource/' though

Comment (rlenglet) [reply pkg/envoy/resources.go]: 👍🏻

**Review by ianvernon — APPROVED** [review]
Approving for CLI.

**Comment by joestringer** [comment]
Previous CI run was successful, I just rebased and updated strings per @rlenglet 's comments.

**Review by rlenglet — APPROVED** [review]

**Comment by jrajahalme** [comment]
Backport #3856

**Pull request merged**